clusterctl/examples/openstack/provider-components.yaml.template

```diff
@@ -0,0 +1,292 @@
+        volumeMounts:
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+        command:
+        - "./controller-manager"
+        args:
+        - --kubeconfig=/etc/kubernetes/admin.conf
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      - name: openstack-machine-controller
+        image: gcr.io/openstack-cluster-api/openstack-machine-controller:0.0.1
+        volumeMounts:
+          - name: config
+            mountPath: /etc/kubernetes
+          - name: certs
+            mountPath: /etc/ssl/certs
+          - name: sshkeys
+            mountPath: /etc/sshkeys
+          - name: machine-setup
+            mountPath: /etc/machinesetup
+          - name: cloudConfig
+            mountPath: /etc/cloud
+          - name: kubeadm
+            mountPath: /usr/bin/kubeadm
+        command:
+        - "./openstack-machine-controller"
+        args:
+        - --kubeconfig=/etc/kubernetes/admin.conf
+        resources:
+          requests:
+            cpu: 100m
+            memory: 20Mi
+          limits:
+            cpu: 100m
+            memory: 30Mi
+      volumes:
+      - name: config
+        hostPath:
+          path: /etc/kubernetes
+      - name: certs
+        hostPath:
+          path: /etc/ssl/certs
+      - name: sshkeys
+        secret:
+          secretName: machine-controller-sshkeys
+          defaultMode: 256
+      - name: cloudConfig
+        secret:
+          secretName: cloud-config
+      - name: machine-setup
+        configMap:
+          name: machine-setup
+      - name: kubeadm
+        hostPath:
+          path: /usr/bin/kubeadm
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: machine-controller-sshkeys
+  namespace: default
+data:
+  user: $MACHINE_CONTROLLER_SSH_USER
+  private: $MACHINE_CONTROLLER_SSH_PRIVATE
+  public: $MACHINE_CONTROLLER_SSH_PUBLIC
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: cloud-config
+  namespace: default
+data:
+  config_config.yaml: $OPENSTACK_CLOUD_CONFIG
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: machine-setup
+data:
+  machine_setup_configs.yaml: |-
+    items:
+    - machineParams:
+      - os: ubuntu-1604-lts
+        roles:
+        - Master
+        versions:
+          kubelet: 1.9.4
+          controlPlane: 1.9.4
+          containerRuntime:
+            name: docker
+            version: 1.12.0
+      metadata:
+        startupScript: |
+          set -e
+          set -x
+          (
+          ARCH=amd64
+          curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key add -
+          touch /etc/apt/sources.list.d/kubernetes.list
+          sh -c 'echo "deb http://apt.kubernetes.io/ kubernetes-xenial main" > /etc/apt/sources.list.d/kubernetes.list'
```

Comment:
when we prompt the user for the image name in glance, we should probably tell them that it should be a 16.04/xenial image or this stuff would not work. right?